### ISSUE TITLE
Warning and extra text when using Markdown as mainpage

### ISSUE THREAD

**Issue by rveerd** [doxygen/doxygen#8078]
**Describe the bug**

I have a default-generated ./doc/Doxyfile with the following changes:

```
INPUT                  = ../README.md
USE_MDFILE_AS_MAINPAGE = README.md
```

And a minimal ./README.md file:

```
README
======

Readme
```

Doxyfile is located in ./doc, README.md is located in . and I run Doxygen from ./build.

This gives the following warnings:

```
$ cd build
$ doxygen ../doc/Doxyfile
...
Generating index page...
/home/user@vb.local/sources/doxygen-bug/README.md:2: warning: Found unknown command '@vb'
/home/user@vb.local/sources/doxygen-bug/README.md:2: warning: Found unknown command '@vb'
```

Also, index.html contains some extra text I do not expect:

```
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">README </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p><a class="anchor" id="md__home_user"></a>@vb.local_sources_doxygen-bug_README</p>
<p>Readme </p>
</div></div><!-- PageDoc -->
```

**Expected behavior**
I do not expect the warnings nor the extra text. I suspect the ampersand (@) in the path causes the issue.

**Version**
This did not happen before, I think that was with 1.8.16. First seen with Doxygen 1.8.20. On RHEL 8 64-bit with Doxygen build from source.

**Additional context**
This does not happen when everything is in the same directory and Doxygen is run from this directory.

**Comment by albert-github** [comment]
When running with `-d markdown` I see:
```
@mainpage README\ilinebr @anchor md_D__speeltuin_issue_8078_user@vb.local_README\ilinebr
```

Looks like we have to escape the `@` sign as well.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8079

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
